mne/io/reference.py

```diff
@@ -386,20 +390,36 @@ def set_eeg_reference(inst, ref_channels='average', copy=True,
             return inst, None
 
     inst = inst.copy() if copy else inst
+    # if ch_type is 'auto', search through list to find first reasonable
+    # reference-able channel type.
+    possible_types = ['eeg', 'ecog', 'seeg']
+    if ch_type == 'auto':
+        for type in possible_types:
+            if type in inst:
+                ch_type = type
+                break
+        logger.info('%s channel type selected for re-referencing' % ch_type)
```

Comment:
I would move this up before the `break`, and then use the `for/else` pattern here:
```
else:
    raise ValueError('No EEG, sEEG, or ECoG channels found to rereference')
```
And remove the `if ch_type not ...` below